src/autoscaler/scalingengine/server/scaling_handler.go

```diff
@@ -104,9 +108,30 @@ func (h *ScalingHandler) GetScalingHistories(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
+	if len(orderParam) == 1 {
+		orderStr := strings.ToUpper(orderParam[0])
+		if orderStr == "DESC" {
+			order = db.DESC
+		} else if orderStr == "ASC" {
+			order = db.ASC
+		} else {
+			logger.Error("failed-to-get-order", err, lager.Data{"order": orderParam})
+			handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+				Code:    "Bad-Request",
+				Message: fmt.Sprintf("Incorrect order parameter in query string, the value can only be %s or %s", db.ASC, db.DESC),
+			})
+			return
+		}
+	} else if len(orderParam) > 1 {
+		logger.Error("failed-to-get-order", err, lager.Data{"order": orderParam})
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad-Request",
+			Message: "Incorrect order parameter in query string"})
+		return
```

Comment:
Just a thought, the code can be refactored for error handling, code is duplicated at quite a few spots.